plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoCaseInsensitiveNameMatchingDisabled.java

```diff
@@ -0,0 +1,102 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mongodb.client.MongoClient;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.bson.Document;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.plugin.mongodb.MongoQueryRunner.createMongoClient;
+import static io.trino.plugin.mongodb.MongoQueryRunner.createMongoQueryRunner;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static java.util.Locale.ENGLISH;
+
+public class TestMongoCaseInsensitiveNameMatchingDisabled
+        extends AbstractTestQueryFramework
+{
+    private MongoServer server;
+    private MongoClient client;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        server = new MongoServer();
+        client = createMongoClient(server);
+        return createMongoQueryRunner(server,
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of("mongodb.case-insensitive-name-matching", "false"),
+                ImmutableList.of(),
+                (runner) -> {});
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        server.close();
+        server = null;
+        client.close();
+        client = null;
+    }
+
+    @Test
+    public void testDescribeTable()
+    {
+        String schemaName = "test" + randomNameSuffix();
+        String tableName = "Test" + randomNameSuffix();
+        String schemaTableName = schemaName + "." + tableName.toLowerCase(ENGLISH);
+        client.getDatabase(schemaName).getCollection(tableName).insertOne(new Document("test", "aaa"));
+
+        assertQueryFails(
+                "DESC mongodb." + schemaTableName,
+                "line 1:1: Table 'mongodb." + schemaTableName + "' does not exist");
+    }
+
+    @Test
+    public void testListTables()
+    {
+        List<String> lowerCasedCollections = ImmutableList.of("collection1", "collection2");
+        List<String> upperCasedCollections = ImmutableList.of("Collection3", "COLLECTION4");
+        List<String> allCollections = ImmutableList.<String>builder().addAll(lowerCasedCollections).addAll(upperCasedCollections).build();
+        String schemaName = "test";
+
+        for (String collection : allCollections) {
+            client.getDatabase(schemaName).getCollection(collection).insertOne(new Document("test", "aaa"));
+        }
+
+        assertQuery("SHOW TABLES IN mongodb." + schemaName,
+                "VALUES ('collection1'), ('collection2')");
```

Comment:
Both `Collection3` &`COLLECTION4` should exist. Please take a look at another comment.